packages/next/server/next-server.ts

```diff
@@ -1574,7 +1579,7 @@ export default class Server {
     }
 
     let ssgCacheKey =
-      isPreviewMode || !isSSG || this.minimalMode
+      isPreviewMode || !isSSG || this.minimalMode || opts.supportsDynamicHTML
```

Comment:
small Q: is `!isSSG` equals `opts.supportsDynamicHTML`?

Reply:
It's one of the factors. We could remove the `!isSSG` and `this.minimalMode` check here if we wanted.